core/shared/src/main/scala/coursier/util/Print.scala

```diff
@@ -151,14 +171,18 @@ object Print {
                   Dependency(mod, ver, "", Set.empty, Attributes("", ""), false, false),
                   excluded = true
                 )
-            }
+          }
 
           dependencies.map(Elem(_, excluded = false)) ++
             (if (printExclusions) excluded else Nil)
         }
     }
 
-    if (reverse) {
+
+    if (jsonPrintRequirement.isDefined) {
+      JsonReport(roots.toVector.map(Elem(_, resolution.dependencyArtifacts, excluded = false)), jsonPrintRequirement.get.conflictResolutionForRoots)(_.children, _.reconciledVersionStr, _.requestedVersionStr, _.downloadedFiles)
```

Comment:
You should match on `jsonPrintRequirement` instead of calling `isDefined` / `get` on it, like
```scala
jsonPrintRequirement match {
  case Some(req) =>
    JsonReport(roots.toVector.map(Elem(_, resolution.dependencyArtifacts, excluded = false)), req.conflictResolutionForRoots)(_.children, _.reconciledVersionStr, _.requestedVersionStr, _.downloadedFiles)
  case None =>
    …
}
```